Gordon360/Controllers/RecIM/TeamsController.cs

```diff
@@ -0,0 +1,282 @@
+;       }
+
+        /// <summary>
+        /// Get all team invites of the user
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("invites")]
+        public ActionResult<IEnumerable<TeamExtendedViewModel>> GetTeamInvites()
+        {
+            var username = AuthUtils.GetUsername(User);
+            try
+            {
+                var teamInvites = _teamService.GetTeamInvitesByParticipantUsername(username);
+                return Ok(teamInvites);
+            }
+            catch(Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets number of games a participant has participated in for a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="username"></param>
+        /// <returns>number of games a participant has attended for a team</returns>
+        [HttpGet]
+        [Route("{teamID}/attendance")]
+        public async Task<ActionResult<int>> NumberOfGamesParticipatedByParticipant(int teamID, [FromBody] string username)
+        {
+            var res = _teamService.NumberOfGamesParticipatedByParticipant(teamID, username);
+            return Ok(res);
+        }
+        /// <summary>
+        /// Accept one specified team invite and true delete others from the same activity if there's any
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="response"></param>
+        /// <returns>The accepted TeamInviteViewModel</returns>
+        [HttpPatch]
+        [Route("{teamID}/invite/status")]
+        public async Task<ActionResult<ParticipantTeamViewModel?>> AcceptTeamInvite(int teamID, [FromBody]string response)
+        {
+            var username = AuthUtils.GetUsername(User);
+            try
+            {
+                var invite = _teamService.GetParticipantTeam(teamID, username);
+                if (invite is null)
+                    return NotFound("You were not invited by this team.");
+                if (username != invite.ParticipantUsername)
+                    return Forbid($"You are not permitted to accept invitations for another participant.");
+
+                switch (response)
+                {
+                    case "accepted":
+                        var joinedParticipantTeam = await _teamService.UpdateParticipantRoleAsync(invite.TeamID,
+                            new ParticipantTeamUploadViewModel { Username = username, RoleTypeID = 3 }
+                            );
+                        return CreatedAtAction("AcceptTeamInvite", joinedParticipantTeam);
+                    case "rejected":
+                        await _teamService.DeleteParticipantTeamAsync(invite.TeamID, username);
+                        return Ok();
+                    default:
+                        return BadRequest("Request does not specify valid invite action");
+                }
```

Comment:
This was something that I asked @silasw2, the argument to keep it as a string would be if there were other response types.

Reply:
That might be thinking a little bit too far ahead. It will be pretty easy to change the parameter type at that point if need be. Obv not a blocking change.